### PR TITLE
Mixin3D.shell: removed deprecated OCCT method

### LINKED ISSUE #649: Deprecated method used in Mixin3D.shell
From https://old.opencascade.com/doc/occt-7.4.0/refman/html/class_b_rep_offset_a_p_i___make_thick_solid.html#a1b559bffad21d24f792772e87625fecb, the constructor used in `Mixin3D.shell` is deprecated, and needs to be replaced with an empty constructor and `MakeThickSolidByJoin` method (I think?).

https://github.com/CadQuery/cadquery/blob/e4ed8816171e429afc511602049e74beef388640/cadquery/occ_impl/shapes.py#L2023-L2083

It's a long shot, but perhaps this will help with #638, where we struggled to shell a shape that had been transformed.

### PR DESCRIPTION
**Pull request by marcus7070**
Will close #649.

I also shortened the code a bit, there was a few duplicated calls.

### CODE REVIEW

File: cadquery/occ_impl/shapes.py [hunk 1]
```diff
@@ -2371,45 +2373,38 @@ def shell(
         }
 
         occ_faces_list = TopTools_ListOfShape()
+        shell_builder = BRepOffsetAPI_MakeThickSolid()
 
-        if faceList:
```
Comment (jmwright) [inline cadquery/occ_impl/shapes.py:2375]: Why did you remove this check to make sure faceList is not None? I see that same check used below.

**Review by jmwright — APPROVED** [review]
I just had the one question. Other than that it looks good. Thanks @marcus7070

File: cadquery/occ_impl/shapes.py (hunk 1 above)
Comment (marcus7070) [reply cadquery/occ_impl/shapes.py:2375]: I was going by the type hinting on the signature (`faceList: Iterable[Face],`), so I didn't think `None` was an option.
But you're right, the way that code was originally written it would work with `None` so I should probably change the type hinting to `Optional[Iterable[Face]]` and handle it.

Comment (marcus7070) [reply cadquery/occ_impl/shapes.py:2375]: e41b0300138350e50870292fedf63b606587e68a should take care of this.

File: cadquery/occ_impl/shapes.py [hunk 2]
```diff
@@ -2351,18 +2351,20 @@ def chamfer(
 
     def shell(
         self: Any,
-        faceList: Iterable[Face],
+        faceList: Optional[Iterable[Face]],
```
Comment (adam-urbanczyk) [inline cadquery/occ_impl/shapes.py:2354]: Empty list evaluates to false - not need to make this Optional.

File: cadquery/occ_impl/shapes.py [hunk 3]
```diff
@@ -2371,45 +2373,39 @@ def shell(
...
+            thickness,
+            tolerance,
+            Intersection=True,
+            Join=kind_dict[kind],
+        )
+        shell_builder.Build()
 
-            shell_builder.Build()
-            rv = shell_builder.Shape()
+        if faceList:
```
Comment (adam-urbanczyk) [inline cadquery/occ_impl/shapes.py:2392]: NB: if [] == if False

File: cadquery/occ_impl/shapes.py (hunk 2 above)
Comment (marcus7070) [reply cadquery/occ_impl/shapes.py:2354]: Indeed, but that wasn't why I added the Optional. The way this function was originally written it would have worked with `faceList=None`. Later it was typehinted as Iterable.

The way I wrote it in my first commit `faceList=None` would have raised a TypeError at `for f in faceList:`. I thought it safer to handle `None` incase there was any user doing it already.

Happy to just go back to my first commit if you are.

Comment (adam-urbanczyk) [reply cadquery/occ_impl/shapes.py:2354]: OK, sounds good

**Review by adam-urbanczyk — APPROVED** [review]
